frame/balances/src/lib.rs

```diff
@@ -786,7 +786,7 @@ pub mod pallet {
 					// Gah!! We have a non-zero reserve balance but no provider refs :(
 					// This shouldn't practically happen, but we need a failsafe anyway: let's give
 					// them enough for an ED.
-					a.free = a.free.min(Self::ed());
+					a.free = a.free.max(Self::ed());
```

Comment:
added warn log

Reply:
There is [defensive_max](https://github.com/paritytech/substrate/blob/e374a33fe1d99d59eb24a08981090bdb4503e81b/frame/support/src/traits/misc.rs#L536) for these use-cases.

Reply:
Or you can use the `defensive!` macro which panics in tests and does print an error so that we see it in the runtime monitoring. Anyway, i thought about using `defensive_max` but it is no good here since you want the warning in both cases.